spot-client/src/spot-tv/ui/components/exit-electron/ExitVerification.js

```diff
@@ -3,20 +3,98 @@ import React from 'react';
 import { withTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
 
-import { getProductName } from 'common/app-state';
-import { Button, StatusOverlay } from 'common/ui';
+import { addNotification, getProductName } from 'common/app-state';
+import { logger } from 'common/logger';
+import { Button, LoadingIcon, StatusOverlay } from 'common/ui';
+
+import { fetchExitPassword } from 'spot-tv/app-state';
+
+const VIEW_LOADING = 'loading';
+const VIEW_ENTER_PASSWORD = 'password';
+const VIEW_EXIT_CONFIRMATION = 'confirmation';
 
 /**
- * Displays an overlay to get confirmation before exiting the application.
+ * The component will first try to fetch the exit password and if it's defined it will ask the user to enter it,
+ * otherwise will ask for the exit action confirmation.
  */
 export class ExitVerification extends React.Component {
     static propTypes = {
         onCancel: PropTypes.func,
+        onPasswordInvalid: PropTypes.func,
+        onSomethingWentWrong: PropTypes.func,
         onVerification: PropTypes.func,
         productName: PropTypes.string,
         t: PropTypes.func
     };
 
+    /**
+     * A constructor.
+     *
+     * @param {Object} props - React props.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            viewToDisplay: VIEW_LOADING
+        };
+
+        this._onPasswordSubmit = this._onPasswordSubmit.bind(this);
+        this._onInputChanged = this._onInputChanged.bind(this);
+    }
+
+    /**
+     * Shows the loading screen while the exit password is being checked.
+     *
+     * @returns {void}
+     */
+    componentDidMount() {
+        fetchExitPassword()
+            .then(exitPassword => {
+                if (exitPassword) {
+                    this.setState({
+                        viewToDisplay: VIEW_ENTER_PASSWORD,
+                        exitPassword
+                    });
+                } else {
+                    this.setState({
+                        viewToDisplay: VIEW_EXIT_CONFIRMATION
+                    });
+                }
+            }, () => {
+                this.props.onSomethingWentWrong();
+                this.props.onCancel();
```

Comment:
I'm not sure about the UX of preventing close if there is an error fetching the password. Maybe it should show the exit confirmation instead? That makes the password a bit less safe but the feature already has workarounds.

Reply:
This is not UX for failed fetch password, but for the case that is supposed to never happen. The fetchExitPassword should never reject because on failed request will return 'undefined' or the most recently cached value. Maybe it should throw and crash the app instead?

Reply:
I will change that to throw